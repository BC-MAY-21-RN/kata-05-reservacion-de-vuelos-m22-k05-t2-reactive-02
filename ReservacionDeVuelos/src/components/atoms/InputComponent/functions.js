import styles from './style';

const func = {
  textstyle: title => {
    return {
      ...styles.titleInput,
      ...(title === 'First Name' ? styles.font18 : styles.font16),
    };
  },
  inputstyle: text => {
    return {
      ...styles.input,
      ...(text ? styles.blue : styles.gray),
    };
  },
};

export default func;
