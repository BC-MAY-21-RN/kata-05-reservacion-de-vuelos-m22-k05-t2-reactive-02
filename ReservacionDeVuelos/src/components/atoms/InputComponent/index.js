import React, {useState} from 'react';
import {View, Text, TextInput, TouchableOpacity} from 'react-native';
import texts from '../../../consts/text';
import VisibleSVG from '../../../assets/icons/visible.svg';
import VisibleBlueSVG from '../../../assets/icons/visibleblue.svg';
import styles from './style';
import func from './functions';
import globalstyles from '../../../consts/globalstyles';

const InputTitle = ({title, input, alert}) => {
  return (
    <Text style={func.textstyle(title)}>
      {title}{' '}
      {title !== 'First Name' ? (
        <Text style={!alert ? {} : styles.textpink}>
          {!alert ? '*' : texts.register[input + 'warning']}
        </Text>
      ) : null}
    </Text>
  );
};

export default function InputComponent({
  title,
  changeForm,
  input,
  text,
  alert,
}) {
  const [iconisSelect, setIconIsSelect] = useState(true);
  return (
    <View style={globalstyles.container}>
      <InputTitle title={title} input={input} alert={alert} />
      <TextInput
        secureTextEntry={title === 'Password' ? iconisSelect : false}
        style={func.inputstyle(text)}
        onChange={(e, a) => changeForm(e.nativeEvent.text, input)}
      />
      {title === 'Password' ? (
        <TouchableOpacity
          onPress={() => setIconIsSelect(!iconisSelect)}
          style={styles.icon}>
          {iconisSelect ? <VisibleSVG /> : <VisibleBlueSVG />}
        </TouchableOpacity>
      ) : null}
      {title === 'Password' ? (
        <Text style={styles.textBottom}>{texts.register.textinput3}</Text>
      ) : null}
    </View>
  );
}
