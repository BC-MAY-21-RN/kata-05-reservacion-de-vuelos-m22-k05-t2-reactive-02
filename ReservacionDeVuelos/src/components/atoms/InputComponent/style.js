import {StyleSheet, Dimensions} from 'react-native';
import colors from '../../../consts/colors';

const styles = StyleSheet.create({
  font18: {fontSize: 18},
  font16: {fontSize: 16},
  titleInput: {
    marginLeft: 20,
    marginBottom: '1.23%',
    color: colors.titleInputs,
  },
  input: {
    ...{
      width: Dimensions.get('screen').width - 40,
      borderWidth: 1,
      fontSize: 17,
      marginLeft: 20,
      borderRadius: 2,
    },
    ...(Dimensions.get('screen').height <= 640 ? {height: 40} : {height: 50}),
  },
  textBottom: {
    marginLeft: 20,
    fontSize: 13,
    marginTop: '1.2%',
    color: colors.passwordtextvalidation,
  },
  icon: {
    ...{position: 'absolute', right: 35},
    ...(Dimensions.get('screen').height <= 640 ? {top: 40} : {top: 45}),
  },
  blue: {
    borderColor: colors.bluetitle,
  },
  gray: {
    borderColor: colors.InputBorder,
  },
  colorblue: {
    color: colors.bluetitle,
  },
  textpink: {
    color: colors.pink,
  },
});

export default styles;
