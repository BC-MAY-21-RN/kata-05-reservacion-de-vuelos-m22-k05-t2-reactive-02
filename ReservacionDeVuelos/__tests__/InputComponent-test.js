import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import InputComponent from '../src/components/atoms/InputComponent';

describe('Tests InputComponent component', () => {
  it('InputComponent snapshot', () => {
    const renderInputComponent = renderer
      .create(
        <InputComponent
          alert={false}
          changeForm={null}
          input={'input0'}
          text={null}
          title={'prueba'}
        />,
      )
      .toJSON();
    expect(renderInputComponent).toMatchSnapshot();
  });
});
