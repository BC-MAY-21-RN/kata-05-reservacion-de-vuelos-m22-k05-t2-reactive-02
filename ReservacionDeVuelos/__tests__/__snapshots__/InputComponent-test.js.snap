// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests InputComponent component InputComponent snapshot 1`] = `
<View
  style={
    Object {
      "marginTop": 18,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#333333",
        "fontSize": 16,
        "marginBottom": "1.23%",
        "marginLeft": 20,
      }
    }
  >
    prueba
     
    <Text
      style={Object {}}
    >
      *
    </Text>
  </Text>
  <TextInput
    onChange={[Function]}
    secureTextEntry={false}
    style={
      Object {
        "borderColor": "#5E656F",
        "borderRadius": 2,
        "borderWidth": 1,
        "fontSize": 17,
        "height": 50,
        "marginLeft": 20,
        "width": 710,
      }
    }
  />
</View>
`;
